packages/ui/src/memberships/modals/TransferInviteModal/TransferInviteFormModal.tsx

```diff
@@ -17,15 +21,46 @@ interface Props {
   member?: Member
 }
 
+const formDefaultValues = {
+  from: undefined,
+  amount: undefined,
+  to: undefined,
+}
+
 export function TransferInviteFormModal({ onClose, onAccept, icon, member }: Props) {
-  const [from, setFrom] = useState<Member | undefined>(member)
-  const [to, setTo] = useState<Member>()
-  const [amount, setAmount] = useNumberInput(0)
-  const filterRecipient = useCallback(filterMember(from), [from])
+  const filterRecipient = useCallback(filterMember(member), [member])
+  const TransferInviteSchema = Yup.object().shape({
+    from: AccountSchema.required('This field is required'),
+    amount: Yup.number()
+      .max(member?.inviteCount ?? 0, `You only have ${member?.inviteCount} invites left.`)
+      .required('This field is required'),
+    to: AccountSchema.required('This field is required'),
+  })
 
-  const isAmountValid = !from || parseInt(amount) <= from.inviteCount
-  const isDisabled = !amount || !isAmountValid || !from || !to
-  const isShowError = amount && !isAmountValid
+  const form = useForm<InviteMembershipFormFields>({
+    resolver: useYupValidationResolver(TransferInviteSchema),
+    mode: 'onChange',
+    defaultValues: formDefaultValues,
+  })
+
+  useEffect(() => {
+    return form.setValue('from', member)
+  }, [member])
+
+  const [from, to] = form.watch(['from', 'to'])
+  const hasError = enhancedHasError(form.formState.errors)
+  const getErrorMessage = enhancedGetErrorMessage(form.formState.errors)
+
+  const setReceiverAccount = useCallback(
```

Comment:
It would be good to consider keeping onChanges with same approach. RN `from` is handled inline and `to` in useCallback